src/fft.jl

```diff
@@ -100,6 +99,51 @@ function compute_fft_size_precise(lattice::AbstractMatrix{T}, Ecut, kpoints;
     fft_size
 end
 
+# Main entry point from pwbasis. Uses the above functions to find out
+# the correct fft_size according to user specification
+function compute_final_fft_size(model::Model{T}, fft_size, Ecut, supersampling,
+                                variational, optimize_fft_size, kcoords) where {T}
+    if variational
+        @assert Ecut > 0
+        if fft_size === nothing
+            fft_size = compute_fft_size(model, Ecut;
+                                        supersampling=supersampling)
```

Comment:
If you want we can drop the supersampling parameter in this function (and in the PlaneWaveBasis constructors). I have not really been using it in the wild and always called `compute_fft_size` explicitly. I think this is fine, because it does represent an advanced use case.

Reply:
I think let's keep it for now